src/compiler/checker.ts

```diff
@@ -38932,6 +38944,16 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const saveCurrentNode = currentNode;
         currentNode = node;
         instantiationCount = 0;
+        // #56013
+        if (node.kind === SyntaxKind.CallExpression && chooseOverloadRecursionLevel >= 0) {
+            let setOfNode: Set<Node> | undefined;
+            if (chooseOverloadRecursionLevel >= 0 && (setOfNode = chooseOverloadFlushNodesSignaturesReq[chooseOverloadRecursionLevel])) {
+                if (!setOfNode.has(node)) {
+                    getNodeLinks(node).resolvedSignature = undefined;

```

Comment:
The request is read and `resolvedSignature` is cleared, but only once per overload loop.